templates/master/00-master/baremetal/files/baremetal-haproxy.yaml

```diff
@@ -0,0 +1,185 @@
+filesystem: "root"
+mode: 0644
+path: "/etc/kubernetes/manifests/haproxy.yaml"
+contents:
+  inline: |
+    kind: Pod
+    apiVersion: v1
+    metadata:
+      name: haproxy
+      namespace: openshift-kni-infra
+      creationTimestamp:
+      deletionGracePeriodSeconds: 65
+      labels:
+        app: kni-infra-api-lb
+    spec:
+      volumes:
+      - name: resource-dir
+        hostPath:
+          path: "/etc/kubernetes/static-pod-resources/haproxy"
+      - name: kubeconfig
+        hostPath:
+          path: "/etc/kubernetes/kubeconfig"
+      - name: run-dir
+        empty-dir: {}
+      - name: conf-dir
+        empty-dir: {}
+      - name: chroot-host
+        hostPath:
+          path: "/"
+      initContainers:
+      - name: clusterrc-generation
+        image: quay.io/openshift-metalkube/kubeconfig-extractor:latest
+        command:
+        - "/usr/bin/kubeconfig-extractor"
+        args:
+        - "/etc/kubernetes/kubeconfig"
+        - "/etc/kubernetes/static-pod-resources/clusterrc"
+        resources: {}
+        volumeMounts:
+        - name: resource-dir
+          mountPath: "/etc/kubernetes/static-pod-resources"
+        - name: kubeconfig
+          mountPath: "/etc/kubernetes/kubeconfig"
+      - name: render-haproxy-cfg
+        image: quay.io/openshift/origin-node:latest
```

Comment:
This image (used in many places in this PR) doesn't seem like it's actually part of our `image-references`.  `quay.io/openshift/origin-cli:latest` is however already used by the etcd quorum guard, and if you just want to run some shell script, let's use that.